src/eos-google-chrome-installer

```diff
@@ -160,36 +158,44 @@ class GoogleChromeInstaller:
         return connectivity == NetworkManager.ConnectivityState.FULL and state == NetworkManager.State.CONNECTED_GLOBAL
 
     def _wait_for_network_connectivity(self):
-        logging.info("Checking network connectivity...")
-        if self._is_connected_state():
-            logging.info("Network connected")
-            return
-
-        logging.info("Not connected to any network, wait for connection")
-
-        loop = GLib.MainLoop()
-
-        def handler(*args):
-            if len(args) < 1:
+        def _response_received(connection, sender, path, interface, signal, params, loop):
+            if len(params) < 2:
                 return
-            state = args[0]
-            if not isinstance(state, dbus.Dictionary):
+
+            state = params[1]
+            if not isinstance(state, dict):
                 return
+
             if state.get('Connectivity', None) != NetworkManager.ConnectivityState.FULL:
                 logging.info("Not connected to the internet, still waiting")
                 return
+
             if state.get('State', None) != NetworkManager.State.CONNECTED_GLOBAL:
                 logging.info("Not connected to the network, still waiting")
                 return
 
             logging.info("Connected to the network and the internet")
             loop.quit()
 
-        dbus_glib.DBusGMainLoop(set_as_default=True)
-        bus = dbus.SystemBus()
-        bus.add_signal_receiver(handler, dbus_interface='org.freedesktop.NetworkManager',
-                                signal_name='PropertiesChanged')
 
+        logging.info("Checking network connectivity...")
+        if self._is_connected_state():
+            logging.info("Network connected")
+            return
+
+        logging.info("Not connected to any network, wait for connection")
+
+        loop = GLib.MainLoop()
+
+        bus = Gio.bus_get_sync(Gio.BusType.SYSTEM, None)
+        bus.signal_subscribe('org.freedesktop.NetworkManager',
+                             'org.freedesktop.DBus.Properties',
+                             'PropertiesChanged',
+                             None,
+                             None,
+                             Gio.DBusSignalFlags.NONE,
+                             _response_received,
+                             loop)
```

Comment:
I did not look much into this (other than to remove the dbus dependency) so you might be right, although I haven't ever done this from python and I'm not sure how that would work.

Could you share a pointer to some example code or docs?